examples/l3fwd/l3fwd_lpm.c

```diff
@@ -0,0 +1,132 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <inttypes.h>
+#include <sys/types.h>
+#include <string.h>
+#include <sys/queue.h>
+#include <stdarg.h>
+#include <errno.h>
+#include <getopt.h>
+#include <stdbool.h>
+
+#include <rte_debug.h>
+#include <rte_ether.h>
+#include <rte_ethdev.h>
+#include <rte_cycles.h>
+#include <rte_mbuf.h>
+#include <rte_ip.h>
+#include <rte_tcp.h>
+#include <rte_udp.h>
+#include <rte_lpm.h>
+#include <rte_lpm6.h>
+#include <rte_malloc.h>
+
+#include "onvm_nflib.h"
+#include "onvm_pkt_helper.h"
+#include "l3fwd.h"
+
+/* Shared data structure containing host port info. */
+extern struct port_info *ports;
+
+struct ipv4_l3fwd_lpm_route {
+        uint32_t ip; // destination address
+        uint8_t  depth;
+        uint8_t  if_out;
+};
+
+static struct ipv4_l3fwd_lpm_route ipv4_l3fwd_lpm_route_array[] = {
```

Comment:
This should be configurable - users should be able to pass in a JSON that denotes the IP, depth, and outbound port. Maybe let the NF take a filename as an arg that has this info, then parse it in setup_lpm.

Reply:
The last time I spoke to to Prof @twood02 we decided we will use this array instead of JSON format. Maybe we can this feature to a separate PR? Although, I can use your firewall example NF to achieve this here I think.

Reply:
Same for adding specific flows to the flow dir PR.

Reply:
for an example NF this is fine to me.